apps/api/src/services/competition-manager.service.ts

```diff
@@ -317,10 +293,9 @@ export class CompetitionManager {
       throw new Error(`Competition is not active: ${competition.status}`);
     }
 
-    if (this.activeCompetitionCache !== competitionId) {
-      throw new Error(
-        `Competition is not the active one: ${this.activeCompetitionCache}`,
-      );
+    const activeCompetition = await findActive();
```

Comment:
I don't get this logic, we already have a findById and check if it's active above. Looks like duplicated logic

Reply:
ah good catch, fixing